apps/guardian/src-tauri/src/commands.rs

```diff
@@ -59,14 +61,55 @@ const SECRET_SHARE_FILE: &str = "secret-share.json.aes";
 const KEY_SHARE_ENCRYPTION_SECRETS_FILE: &str = "key-share-secrets.json.aes";
 /// The file name of the encrypted secret share for a guardian
 const DECRYPTION_SECRET_STATES: &str = "secret-decryption_states.json.aes";
-
-/// Get the data directory for a guardian account
-fn guardian_data_dir(app_handle: &AppHandle, account: AccountAddress) -> PathBuf {
-    app_handle
+/// The file name of the user configuration
+const USER_CONFIG_FILE: &str = "config.toml";
+
+/// Get the data directory for guardians, creating it if it does not exist.
+fn guardians_data_dir(
+    app_handle: &AppHandle,
+    contract: ContractAddress,
+    network: Network,
+) -> PathBuf {
+    let guardian_dir = app_handle
         .path_resolver()
         .app_data_dir()
         .unwrap() // Path is available as declared in `tauri.conf.json`
-        .join(account.to_string())
+        .join(network.to_string())
+        .join(format!("{}-{}", contract.index, contract.subindex));
+
+    if !guardian_dir.exists() {
+        let _ = std::fs::create_dir_all(&guardian_dir);
+    }
+    guardian_dir
+}
+
+/// Get the data directory for a guardian account
+fn guardian_data_dir(
+    app_handle: &AppHandle,
+    account: AccountAddress,
+    contract: ContractAddress,
+    network: Network,
```

Comment:
In case that avoids duplication, we could pass the `user_config` here and then get the `contract`, and `network` out of it inside the function.

Reply:
Sure that makes sense, would also be a good way to make it easier to add to the user config down the line if needed. Will adapt 😄